storage/types.go

```diff
@@ -83,7 +83,7 @@ func NewNodeIDWithPrefix(prefix uint64, prefixLenBits, nodeIDLenBits, maxLenBits
 // index
 func NewNodeIDForTreeCoords(depth int64, index int64, maxLenBits int) NodeID {
 	r := NewEmptyNodeID(maxLenBits)
-	for i := 0; index > 0; i++ {
+	for i := len(r.Path) - 1; index > 0; i-- {
```

Comment:
that's just because `index` is modified in the body of the loop, and that hasn't changed.